spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java

```diff
@@ -165,6 +168,10 @@ public void setRemoteDirectoryExpression(Expression remoteDirectoryExpression) {
 	 * @param filter the file list filter.
 	 */
 	public void setFilter(FileListFilter<F> filter) {
+		doSetFilter(filter);
+	}
+
+	public void doSetFilter(FileListFilter<F> filter) {
```

Comment:
but these are in the abstract class and called from the subclasses.

`protected final ...` ??